contracts/HATVault.sol

```diff
@@ -237,30 +243,31 @@ contract HATVault is IHATVault, ERC4626Upgradeable, OwnableUpgradeable, Reentran
     function approveClaim(bytes32 _claimId, uint256 _bountyPercentage) external nonReentrant isActiveClaim(_claimId) {
         Claim memory claim = activeClaim;
         delete activeClaim;
-
-        uint256 _challengePeriod = getChallengePeriod();
-        uint256 _challengeTimeOutPeriod = getChallengeTimeOutPeriod();
         
         // solhint-disable-next-line not-rely-on-time
-        if (block.timestamp >= claim.createdAt + _challengePeriod + _challengeTimeOutPeriod) revert ClaimExpired();
+        if (block.timestamp >= claim.createdAt + claim.challengePeriod + claim.challengeTimeOutPeriod) revert ClaimExpired();
         if (claim.challengedAt != 0) {
             if (
-                msg.sender != getArbitrator() ||
+                msg.sender != claim.arbitrator ||
                 // solhint-disable-next-line not-rely-on-time
-                block.timestamp > claim.challengedAt + _challengeTimeOutPeriod
+                block.timestamp > claim.challengedAt + claim.challengeTimeOutPeriod
             )
                 revert ChallengedClaimCanOnlyBeApprovedByArbitratorUntilChallengeTimeoutPeriod();
             claim.bountyPercentage = _bountyPercentage;
         } else {
             if (
                 // solhint-disable-next-line not-rely-on-time
-                block.timestamp <= claim.createdAt + _challengePeriod
+                block.timestamp <= claim.createdAt + claim.challengePeriod
             ) revert UnchallengedClaimCanOnlyBeApprovedAfterChallengePeriod();
         }
 
         address tokenLock;
 
-        IHATVault.ClaimBounty memory claimBounty = _calcClaimBounty(claim.bountyPercentage);
+        IHATVault.ClaimBounty memory claimBounty = _calcClaimBounty(
+            claim.bountyPercentage,
+            claim.bountyGovernanceHAT,
+            claim.bountyHackerHATVested
```

Comment:
I guess this should also take account the `claim.hacker` and `claim.hackerVested` and `claim.committee` of the current claim instead of using the settings, right?

Perhaps the signature could be `_calcClaimBounty(claim)`

Reply:
Well these can't change during claim so no reason

Reply:
setBountySplit is protected with noActiveClaim